flowauth/backend/flowauth/models.py

```diff
@@ -651,59 +575,60 @@ def make_demodata():
     current_app.logger.debug("Creating demo data.")
     db.drop_all()
     db.create_all()
-    users = [User(username="TEST_USER"), User(username="TEST_ADMIN", is_admin=True)]
-    for user in users:
-        user.password = "DUMMY_PASSWORD"
-
-    # Each user is also a group
-    groups = [
-        Group(name="TEST_USER", user_group=True),
-        Group(name="TEST_ADMIN", user_group=True),
-        Group(name="Test_Group"),
-    ]
-    groups[0].members.append(users[0])
-    groups[1].members.append(users[1])
-    for user in users:
-        groups[2].members.append(user)
-    for x in users + groups:
-        db.session.add(x)
-    # Add some things that you can do
-    with open(Path(__file__).parent / "demo_data" / "api_scopes.txt") as fin:
-        caps = [x.strip() for x in fin.readlines()]
 
     # Add some servers
     test_server = Server(
         name="TEST_SERVER",
-        longest_token_life=2880,
+        longest_token_life_minutes=31 * 24 * 60,
         latest_token_expiry=datetime.datetime.now() + datetime.timedelta(days=365),
     )
-
     db.session.add(test_server)
 
-    # Add some things that you can do on the servers
-    scs = []
-    for cap in caps:
-        sc = ServerCapability(
-            capability=cap,
+    scopes = [
+        reader_scope := Scope(name="get_result", server=test_server),
+        runner_scope := Scope(name="run", server=test_server),
+        example_geo_scope := Scope(
+            name="admin3:dummy_query:dummy_query", server=test_server
+        ),
+    ]
+    db.session.add_all(scopes)
+
+    # Add roles to test server
+    roles = [
+        Role(
+            name="viewer",
             server=test_server,
-            enabled=True,
-            capability_hash=md5(cap.encode()).hexdigest(),
-        )
-        scs.append(sc)
-        db.session.add(sc)
-
-    # Give bob group permissions on test server
-    for sc in test_server.capabilities:
-        gsp = GroupServerPermission(group=groups[0], server_capability=sc)
-        db.session.add(gsp)
-    db.session.add(
-        GroupServerTokenLimits(
-            group=groups[0],
-            longest_life=1440,
-            latest_end=datetime.datetime.now() + datetime.timedelta(days=28),
+            scopes=[reader_scope, example_geo_scope],
+            latest_token_expiry=datetime.datetime.now() + datetime.timedelta(days=30),
+            longest_token_life_minutes=30 * 24 * 60,
```

Comment:
Interesting you went with 30 here instead of 31 as above. Any reason for this?

Reply:
Yep, obliviousness. I'm going to rework the demo data  over the next PR to bring it more into line with 'real' permissions, and I'll tidy up these when I do